### ISSUE TITLE
getMethods also occur NoClassDefFoundError on some devices,maybe add catch around getMethods

### ISSUE THREAD

**Issue by laomengzhu** [greenrobot/EventBus#556]
getMethods also occur NoClassDefFoundError on some devices,maybe add catch around getMethods

```java
Method[] methods; 
try { 
	// This is faster than getMethods, especially when subscribers are fat classes like Activities 
	methods = findState.clazz.getDeclaredMethods(); 
} catch (Throwable th) { 
	try { 
		methods = findState.clazz.getMethods();
	}catch (Throwable th2){
		return;
	}finally { 
		findState.skipSuperClasses = true;
	}
}
```

_Originally posted by @zhangmingxingAtGit in https://github.com/greenrobot/EventBus/issues/149#issuecomment-401004824_

**Comment by greenrobot-team** [comment]
Have a look at the FAQ about NoClassDefFoundError ("A java.lang.NoClassDefFoundError is throw when a subscriber class is registered. What can I do?").
http://greenrobot.org/eventbus/documentation/faq/

Not sure we should handle this case. Can you elaborate what devices are affected and what we should do if the exception occurs?
-ut

**Comment by laomengzhu** [comment]
* Android: 4.4.2
* OS: EMUI 系统 3.0
* Device: Honor 4C 
* Model: CHM-TL00H

https://github.com/greenrobot/EventBus/blob/761731122af864fc4b925e926a6827510a7f07c0/EventBus/src/org/greenrobot/eventbus/SubscriberMethodFinder.java#L157
"clazz.getMethods()" also occur NoClassDefFoundError on some devices
_______

03-07 11:40:08.409  9639  9639 E AndroidRuntime: FATAL EXCEPTION: main
03-07 11:40:08.409  9639  9639 E AndroidRuntime: Process: com.iqiyi.pizza, PID: 9639
03-07 11:40:08.409  9639  9639 E AndroidRuntime: java.lang.NoClassDefFoundError: android/os/PersistableBundle
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at java.lang.Class.getDeclaredMethods(Native Method)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at java.lang.Class.getPublicMethodsRecursive(Class.java:894)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at java.lang.Class.getMethods(Class.java:877)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at org.greenrobot.eventbus.SubscriberMethodFinder.findUsingReflectionInSingleClass(SubscriberMethodFinder.java:159)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at org.greenrobot.eventbus.SubscriberMethodFinder.findUsingInfo(SubscriberMethodFinder.java:90)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at org.greenrobot.eventbus.SubscriberMethodFinder.findSubscriberMethods(SubscriberMethodFinder.java:64)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at org.greenrobot.eventbus.EventBus.register(EventBus.java:136)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at com.iqiyi.pizza.arch.event.PizzaEventBus.register(PizzaEventBus.kt:25)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at com.iqiyi.pizza.app.base.BaseLifecycleActivity.onCreate(BaseLifecycleActivity.kt:133)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at com.iqiyi.pizza.main.MainActivity.onCreate(MainActivity.kt:158)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:5304)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1090)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2245)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2331)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.app.ActivityThread.access$1000(ActivityThread.java:143)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1244)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:136)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5291)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at java.lang.reflect.Method.invokeNative(Native Method)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Method.java:515)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:849)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:665)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at dalvik.system.NativeStart.main(Native Method)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.os.PersistableBundle" on path: DexPathList[[zip file "/data/app/com.iqiyi.pizza-1.apk", zip file "/data/data/com.iqiyi.pizza/code_cache/secondary-dexes/com.iqiyi.pizza-1.apk.classes2.zip", zip file "/data/data/com.iqiyi.pizza/code_cache/secondary-dexes/com.iqiyi.pizza-1.apk.classes3.zip"],nativeLibraryDirectories=[/data/app-lib/com.iqiyi.pizza-1, /vendor/lib, /system/lib]]
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
03-07 11:40:08.409  9639  9639 E AndroidRuntime: 	... 24 more

**Comment by greenrobot-team** [comment]
OK, `Android: 4.4.2` and `Didn't find class "android.os.PersistableBundle"`: As I said see the FAQ http://greenrobot.org/eventbus/documentation/faq/

**TL;DR:**
Do not override `onCreate(Bundle savedInstanceState, PersistableBundle persistentState)` in your activities. This is only supported for Android 5 and up (API 21).
https://developer.android.com/reference/android/app/Activity.html?hl=in#onCreate(android.os.Bundle,%20android.os.PersistableBundle)

Use `onCreate(Bundle savedInstanceState)` instead.
https://developer.android.com/reference/android/app/Activity.html?hl=in#onCreate(android.os.Bundle)

-ut
